core/src/main/java/de/danoeh/antennapod/core/preferences/SleepTimerPreferences.java

```diff
@@ -69,4 +75,33 @@ public static boolean autoEnable() {
         return prefs.getBoolean(PREF_AUTO_ENABLE, false);
     }
 
+    public static void setAutoEnableTimeBased(boolean autoEnableTimeBased) {
+        prefs.edit().putBoolean(PREF_AUTO_ENABLE_TIME_BASED, autoEnableTimeBased).apply();
+    }
+
+    public static boolean autoEnableTimeBased() {
+        return prefs.getBoolean(PREF_AUTO_ENABLE_TIME_BASED, false);
+    }
+
+    public static void setAutoEnableTimeFrom(int hourOfDay, int minute) {
+        prefs.edit().putInt(PREF_AUTO_ENABLE_TIME_BASED_FROM_MIN, minute).apply();
+        prefs.edit().putInt(PREF_AUTO_ENABLE_TIME_BASED_FROM_HOUR, hourOfDay).apply();
+    }
+
+    public static Pair<Integer, Integer> autoEnableTimeFrom() {
+        Integer minute = prefs.getInt(PREF_AUTO_ENABLE_TIME_BASED_FROM_MIN, 0);
+        Integer hourOfDay = prefs.getInt(PREF_AUTO_ENABLE_TIME_BASED_FROM_HOUR, 0);
+        return new Pair<>(hourOfDay, minute);
+    }
+
+    public static void setAutoEnableTimeTo(int hourOfDay, int minute) {
+        prefs.edit().putInt(PREF_AUTO_ENABLE_TIME_BASED_TO_MIN, minute).apply();
+        prefs.edit().putInt(PREF_AUTO_ENABLE_TIME_BASED_TO_HOUR, hourOfDay).apply();
+    }
+
+    public static Pair<Integer, Integer> autoEnableTimeTo() {
```

Comment:
Maybe create a new class `Time` that implements
* format to string
* compare with `now`

I haven't found a class in the framework that works with time only. They all have fields for the date as well.